### ISSUE TITLE
WP-Rocket prevents Polylang redirect to default language

### ISSUE THREAD

**Issue by webdev69420** [wp-media/wp-rocket#6456]
I use Polylang as a multilingual plugin and WP-Rocket on a clients website.

I've set up Polylang this way:

- Each language has its own slug: https://lecannet-tourisme.fr/fr/ is for the french home-page, etc.
- When the user accesses https://lecannet-tourisme.fr/ he should be redirected to https://lecannet-tourisme.fr/fr/ (the default language, no dynamic redirection based on the browsers language)

This works every time when WP-Rocket is disabled or when I'm connected to the WordPress back-office (and therefore WP-Rocket cache is turned off because I'm logged in).

When WP-Rocket is enabled, it sometimes happens that when a user accesses https://lecannet-tourisme.fr/ he is not redirected. He just sees the same homepage (which loads fine) with the incorrect URL (should be https://lecannet-tourisme.fr/fr/ instead of https://lecannet-tourisme.fr/).

From a user perspective this isn't a big deal (even though it can be confusing too). But for SEO this is a problem to me. https://lecannet-tourisme.fr/ should always redirect to https://lecannet-tourisme.fr/fr/ because I do not want duplicate results of the same page content with 2 different URLs.

Here's how I set up Polylang:

![Screenshot-29-02-2024-17-31-21](https://github.com/wp-media/wp-rocket/assets/44350078/09ef4de2-3649-4f67-80d1-346b6684945b)

Inside WP-Rocket I have cache preloading and link preloading turned on.

To try to temporarily solve this issue myself, I've added these lines to my `.htaccess` file to force the redirect Polylang is supposed to (which doesn't work sometimes, only when WP-Rocket is on):

```
RewriteEngine On
RewriteRule ^$ /fr/ [R=302,L]
```

But even this doesn't seem to work when WP-Rocket is enabled.

I do not have advanced knowledge of how WP-Rocket works on a technical level but I'm under the impression that this is caused because WP-Rocket wrongly cached the homepage under this URL too https://lecannet-tourisme.fr/ and therefore the redirect doesn't happen even though it should. Instead of just caching this URL https://lecannet-tourisme.fr/fr/.

Maybe if there's a way to tell WP-Rocket to never cache https://lecannet-tourisme.fr/ the problem will be solved, but even when I add this URL to the WP-Rocket dashboard to prevent it from preloading this specific URL, the problem still occurs:

![Screenshot-29-02-2024-17-37-38](https://github.com/wp-media/wp-rocket/assets/44350078/aa443266-21cc-4037-9dbe-ed0fbd7624e3)

Normally, I use WPML instead of Polylang to translate my clients websites and this never happened my WPML sites. It also doesn't happen when WP-Rocket is disabled, so I don't think the problem comes from the Polylang plugin. This is why I believe the problem has something to do with WP-Rocket.

Any ideas ?

I can share access to the site or to a pre-production version if needed (WordPress admin access + FTP).

Here's my entire `.htaccess` file if needed for finding the origin of this problem:

```
# BEGIN WP Rocket v3.15.9
# Use UTF-8 encoding for anything served text/plain or text/html
AddDefaultCharset UTF-8
# Force UTF-8 for a number of file formats
<IfModule mod_mime.c>
AddCharset UTF-8 .atom .css .js .json .rss .vtt .xml
</IfModule>

# FileETag None is not enough for every server.
<IfModule mod_headers.c>
Header unset ETag
</IfModule>

# Since we’re sending far-future expires, we don’t need ETags for static content.
# developer.yahoo.com/performance/rules.html#etags
FileETag None

<IfModule mod_alias.c>
<FilesMatch "\.(html|htm|rtf|rtx|txt|xsd|xsl|xml)$">
<IfModule mod_headers.c>
Header set X-Powered-By "WP Rocket/3.15.9"
Header unset Pragma
Header append Cache-Control "public"
Header unset Last-Modified
</IfModule>
</FilesMatch>

<FilesMatch "\.(css|htc|js|asf|asx|wax|wmv|wmx|avi|bmp|class|divx|doc|docx|eot|exe|gif|gz|gzip|ico|jpg|jpeg|jpe|json|mdb|mid|midi|mov|qt|mp3|m4a|mp4|m4v|mpeg|mpg|mpe|mpp|otf|odb|odc|odf|odg|odp|ods|odt|ogg|pdf|png|pot|pps|ppt|pptx|ra|ram|svg|svgz|swf|tar|tif|tiff|ttf|ttc|wav|wma|wri|xla|xls|xlsx|xlt|xlw|zip)$">
<IfModule mod_headers.c>
Header unset Pragma
Header append Cache-Control "public"
</IfModule>
</FilesMatch>
</IfModule>

<IfModule mod_mime.c>
	AddType image/avif                                  avif
    AddType image/avif-sequence                         avifs
</IfModule>
# Expires headers (for better cache control)
<IfModule mod_expires.c>
	ExpiresActive on
	ExpiresDefault                              "access plus 1 month"
	# cache.appcache needs re-requests in FF 3.6 (thanks Remy ~Introducing HTML5)
	ExpiresByType text/cache-manifest           "access plus 0 seconds"
	# Your document html
	ExpiresByType text/html                     "access plus 0 seconds"
	# Data
	ExpiresByType text/xml                      "access plus 0 seconds"
	ExpiresByType application/xml               "access plus 0 seconds"
	ExpiresByType application/json              "access plus 0 seconds"
	# Feed
	ExpiresByType application/rss+xml           "access plus 1 hour"
	ExpiresByType application/atom+xml          "access plus 1 hour"
	# Favicon (cannot be renamed)
	ExpiresByType image/x-icon                  "access plus 1 week"
	# Media: images, video, audio
	ExpiresByType image/gif                     "access plus 4 months"
	ExpiresByType image/png                     "access plus 4 months"
	ExpiresByType image/jpeg                    "access plus 4 months"
	ExpiresByType image/webp                    "access plus 4 months"
	ExpiresByType video/ogg                     "access plus 4 months"
	ExpiresByType audio/ogg                     "access plus 4 months"
	ExpiresByType video/mp4                     "access plus 4 months"
	ExpiresByType video/webm                    "access plus 4 months"
	ExpiresByType image/avif                    "access plus 4 months"
	ExpiresByType image/avif-sequence           "access plus 4 months"
	# HTC files  (css3pie)
	ExpiresByType text/x-component              "access plus 1 month"
	# Webfonts
	ExpiresByType font/ttf                      "access plus 4 months"
	ExpiresByType font/otf                      "access plus 4 months"
	ExpiresByType font/woff                     "access plus 4 months"
	ExpiresByType font/woff2                    "access plus 4 months"
	ExpiresByType image/svg+xml                 "access plus 4 months"
	ExpiresByType application/vnd.ms-fontobject "access plus 1 month"
	# CSS and JavaScript
	ExpiresByType text/css                      "access plus 1 year"
	ExpiresByType application/javascript        "access plus 1 year"
</IfModule>
# Gzip compression
<IfModule mod_deflate.c>
# Active compression
SetOutputFilter DEFLATE
# Force deflate for mangled headers
<IfModule mod_setenvif.c>
<IfModule mod_headers.c>
SetEnvIfNoCase ^(Accept-EncodXng|X-cept-Encoding|X{15}|~{15}|-{15})$ ^((gzip|deflate)\s*,?\s*)+|[X~-]{4,13}$ HAVE_Accept-Encoding
RequestHeader append Accept-Encoding "gzip,deflate" env=HAVE_Accept-Encoding
# Don’t compress images and other uncompressible content
SetEnvIfNoCase Request_URI \
\.(?:gif|jpe?g|png|rar|zip|exe|flv|mov|wma|mp3|avi|swf|mp?g|mp4|webm|webp|pdf)$ no-gzip dont-vary
</IfModule>
</IfModule>

# Compress all output labeled with one of the following MIME-types
<IfModule mod_filter.c>
AddOutputFilterByType DEFLATE application/atom+xml \
		                          application/javascript \
		                          application/json \
		                          application/rss+xml \
		                          application/vnd.ms-fontobject \
		                          application/x-font-ttf \
		                          application/xhtml+xml \
		                          application/xml \
		                          font/opentype \
		                          image/svg+xml \
		                          image/x-icon \
		                          text/css \
		                          text/html \
		                          text/plain \
		                          text/x-component \
		                          text/xml
</IfModule>
<IfModule mod_headers.c>
Header append Vary: Accept-Encoding
</IfModule>
</IfModule>

<IfModule mod_mime.c>
AddType text/html .html_gzip
AddEncoding gzip .html_gzip
</IfModule>
<IfModule mod_setenvif.c>
SetEnvIfNoCase Request_URI \.html_gzip$ no-gzip
</IfModule>

<IfModule mod_rewrite.c>
RewriteEngine On
RewriteBase /
RewriteCond %{HTTPS} on [OR]
RewriteCond %{SERVER_PORT} ^443$ [OR]
RewriteCond %{HTTP:X-Forwarded-Proto} https
RewriteRule .* - [E=WPR_SSL:-https]
RewriteCond %{HTTP:Accept-Encoding} gzip
RewriteRule .* - [E=WPR_ENC:_gzip]
RewriteCond %{REQUEST_METHOD} GET
RewriteCond %{QUERY_STRING} =""
RewriteCond %{HTTP:Cookie} !(wordpress_logged_in_.+|wp-postpass_|wptouch_switch_toggle|comment_author_|comment_author_email_) [NC]
RewriteCond %{REQUEST_URI} !^(/(?:.+/)?feed(?:/(?:.+/?)?)?$|/(?:.+/)?embed/|/(index.php/)?(.*)wp-json(/.*|$))$ [NC]
RewriteCond %{HTTP_USER_AGENT} !^(facebookexternalhit|WhatsApp).* [NC]
RewriteCond "%{DOCUMENT_ROOT}/wp-content/cache/wp-rocket/%{HTTP_HOST}%{REQUEST_URI}/index%{ENV:WPR_SSL}%{ENV:WPR_WEBP}.html%{ENV:WPR_ENC}" -f
RewriteRule .* "/wp-content/cache/wp-rocket/%{HTTP_HOST}%{REQUEST_URI}/index%{ENV:WPR_SSL}%{ENV:WPR_WEBP}.html%{ENV:WPR_ENC}" [L]
</IfModule>
# END WP Rocket

# Polylang sometimes "forgets" to redirect from / to /fr/ so we force this redirect directly inside the .htaccess just in case
RewriteEngine On
RewriteRule ^$ /fr/ [R=302,L]

# Google PageSpeed module settings (if used by server)
<IfModule pagespeed_module>
	Modpagespeed unplugged
	AddOutputFilterByType MOD_PAGESPEED_OUTPUT_FILTER text/html
	ModPagespeedPreserveSubresourceHints On
	ModPagespeedRewriteLevel CoreFilters
	ModPagespeedEnableFilters collapse_whitespace
</IfModule>

# BEGIN WordPress
# Les directives (lignes) entre « BEGIN WordPress » et « END WordPress » sont générées
# dynamiquement, et doivent être modifiées uniquement via les filtres WordPress.
# Toute modification des directives situées entre ces marqueurs sera surchargée.
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
RewriteBase /
RewriteRule ^index\.php$ - [L]
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule . /index.php [L]
</IfModule>

# END WordPress
```

**Comment by webdev69420** [comment]
I just noticed that by putting these lines at the top of the `.htaccess` file instead of below the WP-Rocket lines it solves the problem.

```
RewriteEngine On
RewriteRule ^$ /fr/ [R=302,L]
```

But every time I clear the database using WP-Rocket it pushes its own lines back to the top so this quick fix stops working.

Would it be possible to prevent WP-Rocket from pushing its `.htaccess` lines to the top each time?

**Comment by webdev69420** [comment]
Update: I fixed this issue myself by using the [custom rules WP Rocket plugin](https://docs.wp-rocket.me/article/614-adding-custom-rules-to-the-htaccess-file) to put my custom Rewrites always at the start of my `.htaccess` file.

This solves the issue for me, but I still had to fix it by hand. If anyone could still look into this problem it would be appreciated, so that in the future myself or others won't have to resort to this "dirty fix".

Hope this can help people that have the same issue as me.

Thanks!

My `wp-rocket-htaccess-custom-rules.php` file:
```
<?php
/**
 * Plugin Name: WP Rocket | Add custom htaccess rules
 * Description: Add custom rules to the .htaccess file
 * Author:      WP Rocket Support Team
 * Author URI:  http://wp-rocket.me/
 * License:     GNU General Public License v2 or later
 * License URI: http://www.gnu.org/licenses/gpl-2.0.html
 *
 * Copyright SAS WP MEDIA 2018
 */

namespace WP_Rocket\Helpers\htaccess\custom_rules;

// Standard plugin security, keep this line in place.
defined( 'ABSPATH' ) or die();

/**
 * Add custom rules to the .htaccess file.
 *
 * @author Vasilis Manthos
 * @param  string $marker Block of WP Rocket rules
 * @return string         Extended block of WP Rocket rules
 */
function render_rewrite_rules( $marker ) {

	/*$customHeaders  = '# BEGIN Override default expires' . PHP_EOL;
	$customHeaders  .= '<IfModule mod_expires.c>' . PHP_EOL;
	$customHeaders  .= 'ExpiresActive on' . PHP_EOL;
	$customHeaders  .= 'ExpiresDefault "access plus 0 seconds"' . PHP_EOL;
	$customHeaders  .= '</IfModule>' . PHP_EOL;
	$customHeaders  .= '# END Override default expires' . PHP_EOL . PHP_EOL;*/

	$customHeaders  = '# Polylang sometimes "forgets" to redirect from / to /fr/ so we force this redirect directly inside the .htaccess just in case' . PHP_EOL;
	$customHeaders  = 'RewriteEngine On' . PHP_EOL;
	$customHeaders  = 'RewriteRule ^$ /fr/ [R=302,L]' . PHP_EOL;

	// Prepend custom headers to WP Rocket block.
	$marker = $customHeaders . $marker;

	return $marker;
}

// Use one of those filters to add code before or after WP Rocket rules
add_filter( 'before_rocket_htaccess_rules', __NAMESPACE__ . '\render_rewrite_rules' );
// add_filter( 'after_rocket_htaccess_rules', __NAMESPACE__ . '\render_rewrite_rules' );

/**
 * Updates .htaccess, regenerates WP Rocket config file.
 *
 * @author Caspar Hübinger
 */
function flush_wp_rocket() {

	if ( ! function_exists( 'flush_rocket_htaccess' )
	  || ! function_exists( 'rocket_generate_config_file' ) ) {
		return false;
	}

	// Update WP Rocket .htaccess rules.
	flush_rocket_htaccess();

	// Regenerate WP Rocket config file.
	rocket_generate_config_file();
}
register_activation_hook( __FILE__, __NAMESPACE__ . '\flush_wp_rocket' );

/**
 * Removes customizations, updates .htaccess, regenerates config file.
 *
 * @author Caspar Hübinger
 */
function deactivate() {

	// Remove all functionality added above. Please remove the correct filter.
	remove_filter( 'before_rocket_htaccess_rules', __NAMESPACE__ . '\render_rewrite_rules' );
	// remove_filter( 'after_rocket_htaccess_rules', __NAMESPACE__ . '\render_rewrite_rules' );

	// Flush .htaccess rules, and regenerate WP Rocket config file.
	flush_wp_rocket();
}
register_deactivation_hook( __FILE__, __NAMESPACE__ . '\deactivate' );
```

The new `.htaccess` it generates:
```
# BEGIN WP Rocket v3.15.9
RewriteRule ^$ /fr/ [R=302,L]
# Use UTF-8 encoding for anything served text/plain or text/html
AddDefaultCharset UTF-8
# Force UTF-8 for a number of file formats
<IfModule mod_mime.c>
AddCharset UTF-8 .atom .css .js .json .rss .vtt .xml
</IfModule>

# FileETag None is not enough for every server.
<IfModule mod_headers.c>
Header unset ETag
</IfModule>

# Since we’re sending far-future expires, we don’t need ETags for static content.
# developer.yahoo.com/performance/rules.html#etags
FileETag None

<IfModule mod_alias.c>
<FilesMatch "\.(html|htm|rtf|rtx|txt|xsd|xsl|xml)$">
<IfModule mod_headers.c>
Header set X-Powered-By "WP Rocket/3.15.9"
Header unset Pragma
Header append Cache-Control "public"
Header unset Last-Modified
</IfModule>
</FilesMatch>

<FilesMatch "\.(css|htc|js|asf|asx|wax|wmv|wmx|avi|bmp|class|divx|doc|docx|eot|exe|gif|gz|gzip|ico|jpg|jpeg|jpe|json|mdb|mid|midi|mov|qt|mp3|m4a|mp4|m4v|mpeg|mpg|mpe|mpp|otf|odb|odc|odf|odg|odp|ods|odt|ogg|pdf|png|pot|pps|ppt|pptx|ra|ram|svg|svgz|swf|tar|tif|tiff|ttf|ttc|wav|wma|wri|xla|xls|xlsx|xlt|xlw|zip)$">
<IfModule mod_headers.c>
Header unset Pragma
Header append Cache-Control "public"
</IfModule>
</FilesMatch>
</IfModule>

<IfModule mod_mime.c>
	AddType image/avif                                  avif
    AddType image/avif-sequence                         avifs
</IfModule>
# Expires headers (for better cache control)
<IfModule mod_expires.c>
	ExpiresActive on
	ExpiresDefault                              "access plus 1 month"
	# cache.appcache needs re-requests in FF 3.6 (thanks Remy ~Introducing HTML5)
	ExpiresByType text/cache-manifest           "access plus 0 seconds"
	# Your document html
	ExpiresByType text/html                     "access plus 0 seconds"
	# Data
	ExpiresByType text/xml                      "access plus 0 seconds"
	ExpiresByType application/xml               "access plus 0 seconds"
	ExpiresByType application/json              "access plus 0 seconds"
	# Feed
	ExpiresByType application/rss+xml           "access plus 1 hour"
	ExpiresByType application/atom+xml          "access plus 1 hour"
	# Favicon (cannot be renamed)
	ExpiresByType image/x-icon                  "access plus 1 week"
	# Media: images, video, audio
	ExpiresByType image/gif                     "access plus 4 months"
	ExpiresByType image/png                     "access plus 4 months"
	ExpiresByType image/jpeg                    "access plus 4 months"
	ExpiresByType image/webp                    "access plus 4 months"
	ExpiresByType video/ogg                     "access plus 4 months"
	ExpiresByType audio/ogg                     "access plus 4 months"
	ExpiresByType video/mp4                     "access plus 4 months"
	ExpiresByType video/webm                    "access plus 4 months"
	ExpiresByType image/avif                    "access plus 4 months"
	ExpiresByType image/avif-sequence           "access plus 4 months"
	# HTC files  (css3pie)
	ExpiresByType text/x-component              "access plus 1 month"
	# Webfonts
	ExpiresByType font/ttf                      "access plus 4 months"
	ExpiresByType font/otf                      "access plus 4 months"
	ExpiresByType font/woff                     "access plus 4 months"
	ExpiresByType font/woff2                    "access plus 4 months"
	ExpiresByType image/svg+xml                 "access plus 4 months"
	ExpiresByType application/vnd.ms-fontobject "access plus 1 month"
	# CSS and JavaScript
	ExpiresByType text/css                      "access plus 1 year"
	ExpiresByType application/javascript        "access plus 1 year"
</IfModule>
# Gzip compression
<IfModule mod_deflate.c>
# Active compression
SetOutputFilter DEFLATE
# Force deflate for mangled headers
<IfModule mod_setenvif.c>
<IfModule mod_headers.c>
SetEnvIfNoCase ^(Accept-EncodXng|X-cept-Encoding|X{15}|~{15}|-{15})$ ^((gzip|deflate)\s*,?\s*)+|[X~-]{4,13}$ HAVE_Accept-Encoding
RequestHeader append Accept-Encoding "gzip,deflate" env=HAVE_Accept-Encoding
# Don’t compress images and other uncompressible content
SetEnvIfNoCase Request_URI \
\.(?:gif|jpe?g|png|rar|zip|exe|flv|mov|wma|mp3|avi|swf|mp?g|mp4|webm|webp|pdf)$ no-gzip dont-vary
</IfModule>
</IfModule>

# Compress all output labeled with one of the following MIME-types
<IfModule mod_filter.c>
AddOutputFilterByType DEFLATE application/atom+xml \
		                          application/javascript \
		                          application/json \
		                          application/rss+xml \
		                          application/vnd.ms-fontobject \
		                          application/x-font-ttf \
		                          application/xhtml+xml \
		                          application/xml \
		                          font/opentype \
		                          image/svg+xml \
		                          image/x-icon \
		                          text/css \
		                          text/html \
		                          text/plain \
		                          text/x-component \
		                          text/xml
</IfModule>
<IfModule mod_headers.c>
Header append Vary: Accept-Encoding
</IfModule>
</IfModule>

<IfModule mod_mime.c>
AddType text/html .html_gzip
AddEncoding gzip .html_gzip
</IfModule>
<IfModule mod_setenvif.c>
SetEnvIfNoCase Request_URI \.html_gzip$ no-gzip
</IfModule>

<IfModule mod_rewrite.c>
RewriteEngine On
RewriteBase /
RewriteCond %{HTTPS} on [OR]
RewriteCond %{SERVER_PORT} ^443$ [OR]
RewriteCond %{HTTP:X-Forwarded-Proto} https
RewriteRule .* - [E=WPR_SSL:-https]
RewriteCond %{HTTP:Accept-Encoding} gzip
RewriteRule .* - [E=WPR_ENC:_gzip]
RewriteCond %{REQUEST_METHOD} GET
RewriteCond %{QUERY_STRING} =""
RewriteCond %{HTTP:Cookie} !(wordpress_logged_in_.+|wp-postpass_|wptouch_switch_toggle|comment_author_|comment_author_email_) [NC]
RewriteCond %{REQUEST_URI} !^(/(?:.+/)?feed(?:/(?:.+/?)?)?$|/(?:.+/)?embed/|/(index.php/)?(.*)wp-json(/.*|$))$ [NC]
RewriteCond %{HTTP_USER_AGENT} !^(facebookexternalhit|WhatsApp).* [NC]
RewriteCond "%{DOCUMENT_ROOT}/wp-content/cache/wp-rocket/%{HTTP_HOST}%{REQUEST_URI}/index%{ENV:WPR_SSL}%{ENV:WPR_WEBP}.html%{ENV:WPR_ENC}" -f
RewriteRule .* "/wp-content/cache/wp-rocket/%{HTTP_HOST}%{REQUEST_URI}/index%{ENV:WPR_SSL}%{ENV:WPR_WEBP}.html%{ENV:WPR_ENC}" [L]
</IfModule>
# END WP Rocket

# Google PageSpeed module settings (if used by server)
<IfModule pagespeed_module>
	Modpagespeed unplugged
	AddOutputFilterByType MOD_PAGESPEED_OUTPUT_FILTER text/html
	ModPagespeedPreserveSubresourceHints On
	ModPagespeedRewriteLevel CoreFilters
	ModPagespeedEnableFilters collapse_whitespace
</IfModule>

# BEGIN WordPress
# Les directives (lignes) entre « BEGIN WordPress » et « END WordPress » sont générées
# dynamiquement, et doivent être modifiées uniquement via les filtres WordPress.
# Toute modification des directives situées entre ces marqueurs sera surchargée.
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
RewriteBase /
RewriteRule ^index\.php$ - [L]
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule . /index.php [L]
</IfModule>

# END WordPress
```

**Comment by piotrbak** [comment]
Hello and thanks for creating this issue. I'm not sure if I understand it correctly though, what's the purpose of `https://lecannet-tourisme.fr/`?

**Comment by webdev69420** [comment]
Hi,
The base URL (https://lecannet-tourisme.fr/) should always be redirected to the default language homepage (https://lecannet-tourisme.fr/fr/) by Polylang.

The base URL (https://lecannet-tourisme.fr/) only exists because it is the homepage of the WordPress site. But when using Polylang or WPML I always set those plugins up so that this URL is redirected to /fr/ :

[https://polylang.pro/doc/url-modifications/#front-page-url](https://polylang.pro/doc/url-modifications/#front-page-url)

Therefore, I don't understand why WP Rocket doesn't honor the redirect put in place by Polylang. The base URL should never display anything and always just be a redirect to /fr/.

Sorry, English isn't my primary language, I hope this clears things up a bit!

Basically there's a redirect that should happens by Polylang from / to /fr/ but WP Rocket prevents it and just serves the (working) homepage on the / URL instead of redirectly to /fr/.

**Comment by piotrbak** [comment]
@webdev69420 It depends on how Polylang is set up and how it's handling the redirection. If it's done via PHP, not JS, then the PHP code won't be executed when WP Rocket is used.

My suggestion would be to:
1. Keep the htaccess rule, or
2. Add https://lecannet-tourisme.fr to the `Never Cache URLs` section, so their code is executed

**Comment by Modvlator** [comment]
I think I have a similar issue now with Polylang and WPFORO
Sometimes I do have to turn off and on Polylang to fix the altered backgrounds on some pages with no color on some pages.
https://roatan.day/community/
https://roatan.day/fr/communaute/

**Comment by piotrbak** [comment]
Hello @Modvlator please reach our Support Team for further investigation
